bundles/org.openhab.binding.electroluxappliances/src/main/java/org/openhab/binding/electroluxappliances/internal/handler/ElectroluxAppliancesBridgeHandler.java

```diff
@@ -0,0 +1,191 @@
+                    startAutomaticRefresh();
+
+                });
+            } catch (RuntimeException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void onTokenUpdated(@Nullable String newAccessToken, @Nullable String newRefreshToken) {
+        // Create a new configuration object with the updated tokens
+        Configuration configuration = editConfiguration();
+        configuration.put("accessToken", newAccessToken);
+        configuration.put("refreshToken", newRefreshToken);
+
+        // Update the configuration
+        updateConfiguration(configuration);
+    }
+
+    public Map<String, ApplianceDTO> getElectroluxAppliancesThings() {
+        return electroluxAppliancesThings;
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Set.of(ElectroluxAppliancesDiscoveryService.class);
+    }
+
+    @Override
+    public void dispose() {
+        stopAutomaticRefresh();
+    }
+
+    public @Nullable ElectroluxGroupAPI getElectroluxDeltaAPI() {
+        return api;
+    }
+
+    private boolean refreshAndUpdateStatus() {
+        if (api != null) {
+            if (api.refresh(electroluxAppliancesThings, isCommunicationError)) {
+                getThing().getThings().stream().forEach(thing -> {
+                    ElectroluxAppliancesHandler handler = (ElectroluxAppliancesHandler) thing.getHandler();
+                    if (handler != null) {
+                        handler.update();
+                    }
+                });
+                updateStatus(ThingStatus.ONLINE);
+                isCommunicationError = false;
+                return true;
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                isCommunicationError = true;
+            }
+        }
+        return false;
+    }
+
+    private void startAutomaticRefresh() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            this.refreshJob = scheduler.scheduleWithFixedDelay(this::refreshAndUpdateStatus, 0, refreshTimeInSeconds,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    private void stopAutomaticRefresh() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+        refreshJob = this.instantUpdate;
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+    }
+
+    private synchronized void updateNow() {
+        Future<?> localRef = instantUpdate;
+        if (localRef == null || localRef.isDone()) {
+            instantUpdate = scheduler.schedule(this::refreshAndUpdateStatus, 5, TimeUnit.SECONDS);
```

Comment:
This delay seems not needed or maybe even unwanted as it is a _updateNow_
```suggestion
            instantUpdate = scheduler.schedule(this::refreshAndUpdateStatus, 0, TimeUnit.SECONDS);
```

Reply:
Fixed!